### PR TITLE
Feature/db transactions dao

### PR DESCRIPTION
**Pull request by magdalenagola**

### CODE REVIEW

File: com/codecool/java/dao/DbTransactionsDAO.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        Connection c = pool.getConnection();
+        List<QuestTransaction> unapprovedQuestsList = new ArrayList<>();
+        PreparedStatement ps = c.prepareStatement("SELECT * FROM student_quests WHERE date_approved IS NULL");
+        ResultSet rs = ps.executeQuery();
+        while(rs.next()) {
+            Integer coinsReceived = rs.getInt("coins_received");
+            Date dateAdded = rs.getDate("date_added");
+            Integer questId = rs.getInt("quest_id");
+            Integer userId = rs.getInt("user_id");
+            QuestTransaction questTransaction = new QuestTransaction(questId, userId, dateAdded, coinsReceived);
```
Comment (szczepanTopolski) [inline com/codecool/java/dao/DbTransactionsDAO.java]:
```suggestion
            Transactions questTransaction = new QuestTransaction(questId, userId, dateAdded, coinsReceived);
```
consider use super class type here and at retrun type -> List<Transactions>

Comment (magdalenagola) [reply com/codecool/java/dao/DbTransactionsDAO.java]: good idea, I have changed so where possible

File: com/codecool/java/dao/DbTransactionsDAO.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        }
+        return transactionsList;
+    }
+
+    private void addCardTransaction(CardTransaction cardTransaction) throws SQLException {
+        Connection c = pool.getConnection();
+        PreparedStatement ps = c.prepareStatement("INSERT INTO student_cards(cost, date_bought, user_id) VALUES(?, ?, ?)");
+        ps.setInt(1, cardTransaction.getCost());
+        ps.setInt(2, cardTransaction.getTransactionDate());
+        ps.setInt(4, cardTransaction.getUserId());
```
Comment (e-victoria) [inline com/codecool/java/dao/DbTransactionsDAO.java]: Shouldn't there be: " ps.setInt(**3**, cardTransaction.getUserId());"?

Comment (magdalenagola) [reply com/codecool/java/dao/DbTransactionsDAO.java]: Yes, corrected

**Pull request merged**